### PR TITLE
Various small bug fixes

### PR DESCRIPTION
**Pull request by cholcombe973**
These all came out of testing on a blank ubuntu 18.04 virtual
machine.  Most are minor issues with error logging not showing
me anything useful when failing.

### CODE REVIEW

File: disk-manager/src/main.rs [hunk 1]
```diff
@@ -405,66 +405,80 @@ fn main() {
...
-    ]);
+    let mut loggers: Vec<Box<SharedLogger>> = vec![];
+    if let Some(term_logger) = TermLogger::new(level, Config::default()) {
+        //systemd doesn't use a terminal
+        loggers.push(term_logger);
+    }
+    loggers.push(WriteLogger::new(
+        level,
+        Config::default(),
+        File::create("/var/log/bynar.log").unwrap(),
```
Comment (sdandam) [inline disk-manager/src/main.rs]: This log file location should be a constant or read from a config file?

File: helpers/src/host_information.rs [hunk 2]
```diff
@@ -57,10 +56,8 @@ fn server_type() -> Result<String> {
     debug!("Gathering server type");
     let path = Path::new("/sys/class/dmi/id/product_name");
```
Comment (sdandam) [inline helpers/src/host_information.rs]: WHere is this function called? (my search didn't provide useful leads). Anyway, I am not certain if the paths should be hardcoded. Add a todo/note perhaps?

File: helpers/src/host_information.rs [hunk 3]
```diff
@@ -74,10 +71,8 @@ fn server_serial() -> Result<String> {
     debug!("Checking for serial in /sys/class/dmi/id/product_serial");
     let path_1 = Path::new("/sys/class/dmi/id/product_serial");
```
Comment (sdandam) [inline helpers/src/host_information.rs]: uncertain about hard coded path

File: helpers/src/host_information.rs (hunk 2 above)
Comment (cholcombe973) [reply helpers/src/host_information.rs]: It's used here: https://github.com/Comcast/Bynar/blob/bug_fixes/src/main.rs#L116.  I think that's a special file on linux that gets created if the dmi information is read correctly.  There's a partial implementation of this information here: https://gitlab.redox-os.org/redox-os/dmi/blob/master/src/lib.rs.  I'll check out that code again and see if it does what we need.

File: disk-manager/src/main.rs [hunk 4]
```diff
@@ -65,10 +68,12 @@ fn setup_curve(s: &mut Socket, config_dir: &Path, vault: bool) -> Result<()> {
     // The ubuntu package is linked so this shouldn't fail
     s.set_curve_server(true)?;
     let keypair = zmq::CurveKeyPair::new().map_err(|e| Error::new(ErrorKind::Other, e))?;
+    let hostname =
+        get_hostname().ok_or_else(|| Error::new(ErrorKind::Other, "hostname not found"))?;
+    let key_file = config_dir.join(format!("{}.pem", hostname));
```
[the remark this answers is not in the document]
Comment (cholcombe973) [reply disk-manager/src/main.rs]: I don't think so.  If this errors out then setup_curve will bail before moving on to line 73.

**Pull request merged**